### ISSUE TITLE
Issue in understanding Array/Tuple conversion behaviour in flow documentation

### ISSUE THREAD

**Issue by jinxac** [facebook/flow#3089]
I was reading the flow documentation and I bumped across arity part

```
function array_to_tuple(arr: Array<number>): [number, number] {
  return arr;
}
```

Now according to documentation this should not allowed and documentation says following

**An Array<T> type cannot flow to a tuple, since we don’t know the array’s length**
```
2:   return arr;
            ^ array type. Only tuples and array literals with known elements can flow to
1: function array_to_tuple(arr: Array<number>): [number, number] {
                                                ^ tuple type

Found 1 errors
```

But I was trying something really simple. Following is my code

```
/* @flow */

function array_to_tuple(arr: Array<number>): [number, number] {
  return arr;
}

var a = []
array_to_tuple(a)
```

This seems to work and flow does not throw any errors. This should not work according to the docs.

P.S: I am seeing similar behavior for all cases mentioned in [arity](https://flowtype.org/docs/arrays.html#arity) part of the documentation. None of them are throwing errors, but they should do right?

**Array.Prototype methods**

```
/* @flow */

function mutate_tuple(tup: [number]): void {
  const str = tup.join(', '); // OK
  tup.push(123); // Error
}

mutate_tuple([1,2])
```

Can someone please help????

**Comment by STRML** [comment]
To add to this ticket, I am not seeing this throw an error on any Flow version I tested - and I tested all versions from 0.29.0 onward.

**Comment by jinxac** [comment]
I was trying on version 0.30

```flow version
Flow, a static type checker for JavaScript, version 0.30.0```

**Comment by goodmind** [comment]
Was fixed in Flow v0.38.0
